packages/core/core/src/TransformerRunner.js

```diff
@@ -114,7 +132,7 @@ export default class TransformerRunner {
       if (cacheEntry) {
         let cachedAssets = (
           cacheEntry.initialAssets || cacheEntry.assets
-        ).filter(child => child.hash === asset.hash);
+        ).filter(child => child.hash && child.hash === asset.hash);
```

Comment:
`hash` can be falsy (it could before as well), so make sure we're working with a real value.